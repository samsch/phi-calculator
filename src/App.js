import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';
import phiArray from './phiArray';

const phi = 1.6180339887498948482;

const App = props => (
  <div className="container">
    <div className="app">
      <h1>Phi Calculator</h1>
      <p>Phi = {phi}</p>
      <div className="input-container">
        <button
          className="button"
          type="button"
          onClick={() => props.updateBase(props.base - 1)}
        >
          Decrement
        </button>
        <input
          type="text"
          className="base-input"
          value={props.base}
          onChange={e => props.updateBase(e.target.value)}
        />
        <button
          className="button"
          type="button"
          onClick={() => props.updateBase(props.base + 1)}
        >
          Increment
        </button>
      </div>
      <div className="output-container">
        <ol>
          {phiArray.map((m, i) => (
            <li key={i}>{Math.round(props.base * m * 10000) / 10000}</li>
          ))}
        </ol>
        <ol>
          {phiArray.map((m, i) => (
            <li key={i}>{Math.round(props.base / m * 10000) / 10000}</li>
          ))}
        </ol>
      </div>
    </div>
    <footer>
      Source: <a href="https://github.com/samsch/phi-calculator">https://github.com/samsch/phi-calculator</a>
    </footer>
  </div>
);
App.propTypes = {
  base: PropTypes.number.isRequired,
  updateBase: PropTypes.func.isRequired,
};

export default App;
