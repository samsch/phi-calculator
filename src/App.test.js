/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('The App component', () => {
  test('Should match a snapshot render', () => {
    const props = {
      base: 156,
      updateBase: () => {},
    };

    const renderedApp = renderer.create(<App {...props} />);
    expect(renderedApp.toJSON()).toMatchSnapshot();
  });
});
