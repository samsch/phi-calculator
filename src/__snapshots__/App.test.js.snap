// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App component Should match a snapshot render 1`] = `
<div
  className="container"
>
  <div
    className="app"
  >
    <h1>
      Phi Calculator
    </h1>
    <p>
      Phi = 
      1.618033988749895
    </p>
    <div
      className="input-container"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        Decrement
      </button>
      <input
        className="base-input"
        onChange={[Function]}
        type="text"
        value={156}
      />
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        Increment
      </button>
    </div>
    <div
      className="output-container"
    >
      <ol>
        <li>
          252.4133
        </li>
        <li>
          408.4133
        </li>
        <li>
          660.8266
        </li>
        <li>
          1069.2399
        </li>
        <li>
          1730.0665
        </li>
        <li>
          2799.3064
        </li>
        <li>
          4529.3729
        </li>
        <li>
          7328.6793
        </li>
        <li>
          11858.0523
        </li>
        <li>
          19186.7316
        </li>
      </ol>
      <ol>
        <li>
          96.4133
        </li>
        <li>
          59.5867
        </li>
        <li>
          36.8266
        </li>
        <li>
          22.7601
        </li>
        <li>
          14.0665
        </li>
        <li>
          8.6936
        </li>
        <li>
          5.3729
        </li>
        <li>
          3.3207
        </li>
        <li>
          2.0523
        </li>
        <li>
          1.2684
        </li>
      </ol>
    </div>
  </div>
  <footer>
    Source: 
    <a
      href="https://github.com/samsch/phi-calculator"
    >
      https://github.com/samsch/phi-calculator
    </a>
  </footer>
</div>
`;
